compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -814,6 +814,14 @@ OMR::ARM64::TreeEvaluator::iselectEvaluator(TR::Node *node, TR::CodeGenerator *c
    TR::Register *falseReg = cg->evaluate(falseNode);
    TR::Register *resultReg = trueReg;
 
+   // Internal pointers cannot be handled since we cannot set the pinning array
+   // on the result register without knowing which side of the select will be
+   // taken.
+   if (trueReg->containsInternalPointer() || falseReg->containsInternalPointer())
+      TR_ASSERT(false, "Select node should not have children containing internal pointers.\n");
```

Comment:
In addition to what Andrew pointed out, I'd like to see this assert switched to use the new `TR_ASSERT_FATAL_WITH_NODE` macro so that we get node context printed on failure.